File: Projects/Example/ObjC/RatesObjCExample/Sources/ViewController.m
//
//  ViewController.m
//  RatesObjCExample
//

#import "ViewController.h"

@import SwiftRates;

@interface ViewController () <UIPopoverPresentationControllerDelegate>

@property (weak, nonatomic) IBOutlet UIButton *currencyButton;
@property (weak, nonatomic) IBOutlet UITextField *valueTextField;
@property (weak, nonatomic) IBOutlet UIButton *convertButton;

@end

@implementation ViewController

#pragma mark - UIViewController's methods

- (void)viewDidLoad
{
	[super viewDidLoad];
	
	UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(dismissKeyboard)];
	
	[self.view addGestureRecognizer:tap];
}

- (void)viewWillAppear:(BOOL)animated
{
	[super viewWillAppear:animated];
	
	self.navigationController.navigationBarHidden = YES;
}

#pragma mark - Dismiss keyboard

- (void)dismissKeyboard
{
	[_valueTextField resignFirstResponder];
}

#pragma mark - Actions

- (IBAction)convertButtonAction:(UIButton *)sender
{
	CGFloat rawValue = [_valueTextField.text floatValue];
	
	[SwiftRates showSummaryScreen:rawValue toCurrency:_currencyButton.titleLabel.text inNavigationController:self.navigationController animated:YES];
}

- (IBAction)currencyButtonAction:(UIButton *)sender
{
	[SwiftRates showCurrencyList:sender inViewController:self animated:YES];
}

#pragma mark - UIPopoverPresentationControllerDelegate

- (UIModalPresentationStyle)adaptivePresentationStyleForPresentationController:(UIPresentationController *)controller
{
	return UIModalPresentationNone;
}

@end

File: Projects/Example/Swift/RatesSwiftExample/Sources/ViewController.swift
//
//  ViewController.swift
//  RatesSwiftExample
//
////

import UIKit
import SwiftRates

class ViewController: UIViewController, UIPopoverPresentationControllerDelegate {
	
	// MARK: - Outlets
	
	@IBOutlet weak var currencyButton: UIButton!
	@IBOutlet weak var valueTextField: UITextField!
	@IBOutlet weak var convertButton: UIButton!
	
	// MARK: - UIViewController's methods
	
	override func viewDidLoad() {
		super.viewDidLoad()
		
		let tap = UITapGestureRecognizer(target: self, action: #selector(ViewController.dismissKeyboard))
		self.view.addGestureRecognizer(tap)
	}
	
	override func viewWillAppear(_ animated: Bool) {
		super.viewWillAppear(animated)
		
		self.navigationController?.isNavigationBarHidden = true
	}
	
	// MARK: - Dismiss keyboard
	
	func dismissKeyboard() {
		valueTextField.resignFirstResponder()
	}

	// MARK: - Actions
	
	@IBAction func convertButtonAction(_ sender: UIButton) {
		let rawValue = (valueTextField.text! as NSString).floatValue
		SwiftRates.showSummaryScreen(CGFloat(rawValue), toCurrency: (currencyButton.titleLabel?.text)!, inNavigationController: self.navigationController, animated: true)
	}
	
	@IBAction func currencyButtonAction(_ sender: UIButton) {
		SwiftRates.showCurrencyList(sender, inViewController: self, animated: true)
	}
	
	// MARK: - UIPopoverPresentationControllerDelegate
	
	func adaptivePresentationStyle(for controller: UIPresentationController) -> UIModalPresentationStyle {
		return UIModalPresentationStyle.none
	}

}

